src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        List<MovieSession> movieSessionsFromDb = movieSessionDao.findAll();
+        boolean isMovieSessionExist = movieSessionsFromDb
+                .stream()
+                .anyMatch(ms ->
+                        ms.getCinemaHall().getId().equals(movieSession.getCinemaHall().getId())
+                                && isDateWithinRange(
+                                        movieSession.getStartTime(),
+                                ms.getStartTime(),
+                                ms.getEndTime()));
+        if (isMovieSessionExist) {
+            throw new RuntimeException("Movie session already exists");
+        }
```

Comment:
According to the checklist, you should not get all `MovieSession` from dao and then sort them on service layer in `add()` method. Instead, you should use the dao layer to check for the existence of the movie session directly in the database, which is more efficient.